src/formatDate.js

```diff
@@ -9,6 +9,39 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const dataArr = date.split(fromSeparator);
+  const obj = {};
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    obj[fromFormat[i]] = dataArr[i];
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (const key in obj) {
+      if (toFormat[i] === key) {
+        result.push(obj[key]);
+      }
+    }
+
+    if (toFormat[i] === 'YY') {
+      result.push(obj['YYYY'].split('').slice(2).join(''));
+    }
+
+    if (toFormat[i] === 'YYYY') {
+      if (+obj['YY'] < 30) {
+        result.push(`20${obj['YY']}`);
+      }
+
+      if (+obj['YY'] >= 30) {
+        result.push(`19${obj['YY']}`);
+      }
+    }
```

Comment:
The nested loop inside this for-loop is unnecessary and inefficient. You can directly access the values from the `obj` using the keys specified by `toFormat[i]`. Also, the conditions for converting between `YY` and `YYYY` are not correctly placed. They should be directly related to the format being processed in the current iteration of the loop, not in separate if statements. The conversion should also occur only if the `YY` or `YYYY` keys are actually present in the `toFormat` array.